File: web/src/shared/models/productQuantity.ts
import {Order} from "./order";
import {Product} from "./product";

export class ProductQuantity {
  id: number;
  order: Order;
  product: Product;
  quantity: number;

  constructor(id: number, order: Order, product: Product, quantity: number) {
    this.id = id;
    this.order = order;
    this.product = product;
    this.quantity = quantity;
  }

  static fromJsonObject(jsonObject: any): ProductQuantity {
    return new ProductQuantity(jsonObject.id, jsonObject.order, jsonObject.product, jsonObject.quantity);
  }
}

File: src/main/java/com/picking/blinders/app/models/ProductQuantity.java
package com.picking.blinders.app.models;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;

@Entity
@Table
public class ProductQuantity {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private int quantity;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "order_id")
    private Order order;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.EAGER)
    //@JoinColumn(name = "product_id")
    private Product product;

    public ProductQuantity() {
    }

    public ProductQuantity( int quantity, Order order, Product product) {
        this.quantity = quantity;
        this.order = order;
        this.product = product;
    }

    public Long getId() {
        return id;
    }

    public int getQuantity() {
        return quantity;
    }

    public Order getOrder() {
        return order;
    }

    public Product getProduct() {
        return product;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }

    public void setOrder(Order order) {
        this.order = order;
    }

    public void setProduct(Product product) {
        this.product = product;
    }
}
